cmd/bootstrap/cmd/partner_infos.go

```diff
@@ -0,0 +1,244 @@
+func executeGetProposedTableScript(ctx context.Context, env templates.Environment, flowClient *client.Client) (cadence.Array, error) {
+	proposedTable, err := flowClient.ExecuteScriptAtLatestBlock(ctx, templates.GenerateReturnProposedTableScript(env), []cadence.Value{})
+	if err != nil {
+		return cadence.Array{}, fmt.Errorf("failed to execute the get proposed table script: %w", err)
+	}
+
+	return proposedTable.(cadence.Array), nil
+}
+
+// executeGetNodeInfoScript executes the get node info script
+func executeGetNodeInfoScript(ctx context.Context, env templates.Environment, client *client.Client, nodeID cadence.Value) (cadence.Value, error) {
+	info, err := client.ExecuteScriptAtLatestBlock(ctx, getNodeInfoScript, []cadence.Value{nodeID})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute the get node info script: %w", err)
+	}
+
+	return info, nil
+}
+
+// parseNodeInfo convert node info retrieved from
+func parseNodeInfo(info cadence.Value) (*NodePubInfo, error) {
+	fields := info.(cadence.Struct).Fields
+	role := flow.Role(fields[roleField].(cadence.UInt8))
+	return &NodePubInfo{
+		Role:          role.String(),
+		Address:       string(fields[networkingAddressField].(cadence.String)),
+		NodeID:        string(fields[idField].(cadence.String)),
+		Stake:         fields[tokensStakedField].(cadence.UFix64).String(),
+		NetworkPubKey: string(fields[networkingKeyField].(cadence.String)),
+		StakingPubKey: string(fields[stakingKeyField].(cadence.String)),
+	}, nil
+}
+
+// isFlowNode returns true if the address contains nodes.onflow.org
+func isFlowNode(address string) bool {
+	return strings.Contains(address, flowNodeAddrPart)
+}
+
+// validateANNetworkKey attempts to parse the network key provided for secure client connections
+func validateANNetworkKey(key string) error {
+	b, err := hex.DecodeString(strings.TrimPrefix(key, "0x"))
+	if err != nil {
+		return fmt.Errorf("failed to decode public key hex string: %w", err)
+	}
+
+	_, err = crypto.DecodePublicKey(crypto.ECDSA_P256, b)
+	if err != nil {
+		return fmt.Errorf("failed to decode public key: %w", err)
+	}
+
+	return nil
+}
+
+// writeNodePubInfoFile writes the node-pub-info file
+func writeNodePubInfoFile(info *NodePubInfo) {
+	fileOutputName := fmt.Sprintf(infoFileNameTemplate, info.NodeID)
+	path := filepath.Join(flagOutputDir, nodeInfosPubDir, fileOutputName)
+	writeJSON(path, info)
+}
+
+// writePartnerStakesFile writes the partner stakes file
+func writePartnerStakesFile(partnerStakes PartnerStakesInfo) {
+	path := filepath.Join(flagOutputDir, partnerStakesFileName)
+	writeJSON(path, partnerStakes)
+}
+
+func printNodeCounts(numOfNodesByType map[string]int, totalNumOfPartnerNodes, skippedNodes int) {
+	var builder strings.Builder
+	builder.WriteString(fmt.Sprintf("Number of Flow nodes skipped: %d\n", skippedNodes))
+	builder.WriteString(fmt.Sprintf("Number of Partner nodes: %d\n", totalNumOfPartnerNodes))
+	for role, count := range numOfNodesByType {
```

Comment:
```suggestion
	builder.WriteString(fmt.Sprintf("Total number of Partner nodes: %d\n", totalNumOfPartnerNodes))
	builder.WriteString("Number of Partner nodes by role:\n")
	for role, count := range numOfNodesByType {
```

Being more explicit about what the next line of numbers means

Reply:
https://github.com/onflow/flow-go/pull/1988/commits/c34743b2afef68f94ebeeae98cdca9695a565740